### PR TITLE
Block sueddeutsche search results

### PR DESCRIPTION
**Pull request by mohe2015**
Demo: https://www.sueddeutsche.de/news?search=rechtspopulisten
Fixes https://github.com/dieparteidiepartei/afd-blocker-plugin/issues/4

### CODE REVIEW

File: app/scripts/blocker_general.js [hunk 1]
```diff
@@ -102,6 +102,22 @@ export class Blocker {
...
         addBlocker(nodeConfigurations);
     }
+
+    watchPageForMutations() {
+      var self = this;
+      var mutationObserver = new MutationObserver(function(mutations) {
+        for(var i=0; i<mutations.length; ++i) {
+            // look through all added nodes of this mutation
+            for(var j=0; j<mutations[i].addedNodes.length; ++j) {
+                self.modifyContent(mutations[i].addedNodes[j]);
```
Comment (dieparteidiepartei) [inline app/scripts/blocker_general.js]: Could you deactivate the observer while modifying the content? Maybe you coul collect the changes and run  `modifyContent` as seldom as possible?

Comment (mohe2015) [reply app/scripts/blocker_general.js]: What do you think about that idea (https://github.com/dieparteidiepartei/afd-blocker-plugin/pull/3/commits/df5973b0fb93404acf01d97416226300b7dfdf34) as I don't want to loose any changes?

Comment (mohe2015) [reply app/scripts/blocker_general.js]: Collecting the changes shouldn't make a difference as it only searchs for the match in the added elements

Comment (dieparteidiepartei) [reply app/scripts/blocker_general.js]: I've just tested the code - the problem with this Mutation oberver is, that it fires very often and for every little change. I don't want to create a massive performance problem for users.

I see 2 possible solutions:
1) Don't use a mutation observer and watch for click events on "load more" and "pager" elements
2) "Throttle" the page changes (e.g.: run changes only after 2 seconds or so...)

Comment (mohe2015) [reply app/scripts/blocker_general.js]: All these events are usually fired in one batch (outside of the mutations loop). What do you think about collecting them and then doing all changes at once? I will try it and notify you when I'm done.

Comment (mohe2015) [reply app/scripts/blocker_general.js]: I [changed](https://github.com/dieparteidiepartei/afd-blocker-plugin/pull/3/commits/acca1e3e930a0f5a4ef3189a54d11068625098aa) it now. But I don't think it's way better.

Comment (mohe2015) [reply app/scripts/blocker_general.js]: The XPath was absolute. I [changed](https://github.com/dieparteidiepartei/afd-blocker-plugin/pull/3/commits/b07899bdd3ce2a8f711435699aede30ed643742a) it and I think it should be better now.

Comment (dieparteidiepartei) [reply app/scripts/blocker_general.js]: Okay - looks good now :) I'll comment on a few minor issues

File: app/scripts/blocker_general.js [hunk 2]
```diff
@@ -63,45 +63,73 @@ export class Blocker {
...
         }
         addBlocker(nodeConfigurations);
     }
+
+    watchPageForMutations() {
+      var self = this;
+      var mutationObserver = new MutationObserver(function(mutations) {
+        console.log("++++++++++++++++");
+        let addedNodes = [];
+        for(var i=0; i<mutations.length; ++i) {
```
Comment (dieparteidiepartei) [inline app/scripts/blocker_general.js]: Please use `let` instead of `var`

File: app/scripts/blocker_general.js [hunk 3]
```diff
@@ -63,45 +63,73 @@ export class Blocker {
...
+            // look through all added nodes of this mutation
+            for(var j=0; j<mutations[i].addedNodes.length; ++j) {
```
Comment (dieparteidiepartei) [inline app/scripts/blocker_general.js]: Pleas use `let` instead of `var`

File: app/scripts/blocker_general.js [hunk 4]
```diff
@@ -63,45 +63,73 @@ export class Blocker {
...
+                console.error( 'Error: Document tree modified during iteration ' + e );
+            }
         }
         addBlocker(nodeConfigurations);
     }
+
+    watchPageForMutations() {
+      var self = this;
+      var mutationObserver = new MutationObserver(function(mutations) {
+        console.log("++++++++++++++++");
```
Comment (dieparteidiepartei) [inline app/scripts/blocker_general.js]: can you remove that?

File: app/scripts/blocker_general.js [hunk 5]
```diff
@@ -63,45 +63,73 @@ export class Blocker {
...
+        let addedNodes = [];
+        for(var i=0; i<mutations.length; ++i) {
+            // look through all added nodes of this mutation
+            for(var j=0; j<mutations[i].addedNodes.length; ++j) {
+                if (mutations[i].addedNodes[j].classList && !mutations[i].addedNodes[j].classList.contains('afdcontentblockedoverlay')) {
+                    console.log(mutations[i].addedNodes[j]);
```
Comment (dieparteidiepartei) [inline app/scripts/blocker_general.js]: Can you remove that as well?

File: app/scripts/blocker_general.js [hunk 6]
```diff
@@ -63,45 +63,73 @@ export class Blocker {
...
+            catch (e)
+            {
+                console.error( 'Error: Document tree modified during iteration ' + e );
+            }
         }
         addBlocker(nodeConfigurations);
     }
+
+    watchPageForMutations() {
+      var self = this;
```
Comment (dieparteidiepartei) [inline app/scripts/blocker_general.js]: You can use an arrow function for the callback

File: app/scripts/blocker_general.js [hunk 7]
```diff
@@ -63,45 +63,73 @@ export class Blocker {
...
+      var mutationObserver = new MutationObserver(function(mutations) {
```
Comment (dieparteidiepartei) [inline app/scripts/blocker_general.js]: Please use `let` instead of `var`

File: app/scripts/blocker_general.js (hunk 1 above)
Comment (dieparteidiepartei) [reply app/scripts/blocker_general.js]: And thanks for big PR btw. :)

**Comment by mohe2015** [comment]
Done

**Pull request merged**